src/MultiFormats/VarInt.cs

```diff
@@ -0,0 +1,287 @@
+        /// <param name="value">
+        ///   A non-negative value to write.
+        /// </param>
+        /// <exception cref="NotSupportedException">
+        ///   When <paramref name="value"/> is negative.
+        /// </exception>
+        public static void WriteVarint(this Stream stream, long value)
+        {
+            stream.WriteVarintAsync(value).ConfigureAwait(false).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        ///   Reads a variable integer from the stream. 
+        /// </summary>
+        /// <param name="stream">
+        ///   A varint encoded <see cref="Stream"/>.
+        /// </param>
+        /// <exception cref="EndOfStreamException">
+        ///   When the no bytes exist in the <paramref name="stream"/>.
+        /// </exception>
+        /// <exception cref="InvalidDataException">
+        ///   When the varint value is greater than <see cref="Int32.MaxValue"/>.
+        /// </exception>
+        /// <returns>The integer value.</returns>
+        public static int ReadVarint32(this Stream stream)
+        {
+            return stream.ReadVarint32Async().ConfigureAwait(false).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        ///   Reads a variable integer from the stream. 
+        /// </summary>
+        /// <param name="stream">
+        ///   A varint encoded <see cref="Stream"/>.
+        /// </param>
+        /// <exception cref="EndOfStreamException">
+        ///   When the no bytes exist in the <paramref name="stream"/>.
+        /// </exception>
+        /// <exception cref="InvalidDataException">
+        ///   When the varint value is greater than <see cref="Int64.MaxValue"/>.
+        /// </exception>
+        /// <returns>The integer value.</returns>
+        public static long ReadVarint64(this Stream stream)
+        {
+            return stream.ReadVarint64Async().ConfigureAwait(false).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        ///   Writes the variable integer encoding of the value `to
+        ///   a stream.
+        /// </summary>
+        /// <param name="stream">
+        ///   The <see cref="Stream"/> to write to.
+        /// </param>
+        /// <param name="value">
+        ///   A non-negative value to write.
+        /// </param>
+        /// <param name="cancel">
+        ///   Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException"/> is raised.
+        /// </param>
+        /// <returns>
+        ///   A task that represents the asynchronous operation.
+        /// </returns>
+        /// <exception cref="NotSupportedException">
+        ///   When <paramref name="value"/> is negative.
+        /// </exception>
+        public static async Task WriteVarintAsync(this Stream stream,
+            long value,
+            CancellationToken cancel = default)
+        {
+            if (value < 0) throw new NotSupportedException("Negative values are not allowed for a Varint");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)